File: src/main/java/UI/Scene/MainMenuScene.java
package UI.Scene;

import Logic.Save;
import java.awt.*;
import java.awt.event.ActionEvent;
import javax.swing.*;

public class MainMenuScene implements Scene {
    private final SceneManager sceneManager;

    public MainMenuScene(SceneManager sceneManager) {
        this.sceneManager = sceneManager;
    }

    @Override
    public JMenuBar getMenu() {
        return null;
    }

    @Override
    public Component toComponent() {
        JPanel panel = new JPanel();
        Color backgroundColor = Color.decode("#E8C794");
        // Color backgroundColor = Color.decode("#81B0A4");
        panel.setBackground(backgroundColor);

        ImageIcon logoIcon = new ImageIcon(getClass().getResource("/main_menu_pic.jpg"));
        // ImageIcon logoIcon = new ImageIcon(getClass().getResource("/mainMenu3.jpg"));
        JLabel logoLabel = new JLabel(logoIcon);
        panel.add(logoLabel);

        JButton newGameButton = new JButton("New Game");
        newGameButton.setPreferredSize(new Dimension(150, 50));
        newGameButton.addActionListener(
                (ActionEvent event) -> {
                    Save.createTemporary();
                    sceneManager.transitionToGame();
                });
        panel.add(newGameButton);

        JButton loadButton = new JButton("Load Game");
        loadButton.setPreferredSize(new Dimension(150, 50));
        loadButton.addActionListener(
                (ActionEvent event) -> {
                    Save.load();
                    sceneManager.transitionToGame();
                });
        panel.add(loadButton);

        return panel;
    }

    @Override
    public void refresh() {}

    @Override
    public void onClose() {}
}

File: src/main/kotlin/UI/Scene/MainMenuScene.kt
package UI.Scene;

import Logic.Save;

import javax.swing.*;
import java.awt.*;
import java.awt.event.ActionEvent;

class MainMenuScene(val sceneManager: SceneManager) : Scene {

    override fun getMenu(): JMenuBar? {
        return null
    }

    override fun toComponent(): Component {
        val panel: JPanel  = JPanel()
        val backgroundColor: Color = Color.decode("#E8C794");
        // Color backgroundColor = Color.decode("#81B0A4");
        panel.setBackground(backgroundColor);

        val logoIcon: ImageIcon  = ImageIcon(javaClass.getResource("/main_menu_pic.jpg"));
        // ImageIcon logoIcon = new ImageIcon(getClass().getResource("/mainMenu3.jpg"));
        val logoLabel: JLabel = JLabel(logoIcon);
        panel.add(logoLabel);

        val newGameButton: JButton = JButton("New Game");
        newGameButton.setPreferredSize(Dimension(150, 50));
        newGameButton.addActionListener {event ->
            Save.createTemporary()
            sceneManager.transitionToGame()
        }
        panel.add(newGameButton);

        val loadButton: JButton = JButton("Load Game");
        loadButton.setPreferredSize(Dimension(150, 50));
        loadButton.addActionListener{event->
            Save.load()
            sceneManager.transitionToGame()
        }
        panel.add(loadButton);

        return panel;
    }

   override fun refresh() {}

    override fun onClose() {}
}
